Auth0/CredentialsManager.swift

```diff
@@ -53,11 +53,31 @@ public struct CredentialsManager {
     /// Store credentials instance in keychain
     ///
     /// - Parameter credentials: credentials instance to store
-    /// - Returns: Bool outcome of success
+    /// - Returns: if credentials were stored
     public func store(credentials: Credentials) -> Bool {
         return self.storage.setData(NSKeyedArchiver.archivedData(withRootObject: credentials), forKey: storeKey)
     }
 
+    /// Clear credentials stored in keychain
+    ///
+    /// - Returns: if credentials were removed
+    public func clear() -> Bool {
+        return self.storage.deleteEntry(forKey: storeKey)
+    }
+
+    /// Checks if valid credentials are stored
+    ///
+    /// - Returns: if there are valid and non-expired credentials stored
+    public func hasValid() -> Bool {
+        guard
+            let data = self.storage.data(forKey:self.storeKey),
+            let credentials = NSKeyedUnarchiver.unarchiveObject(with: data) as? Credentials,
+            credentials.accessToken != nil,
+            let expiresIn = credentials.expiresIn
+            else { return false }
+        return expiresIn > Date()
```

Comment:
Having the refresh_token revoked is an edge case. Admin has to do that either manually in the user's dashboard or calling the MGMT API with a well-scoped token. I rather pay the price of a 401 and make the user clear the credentials than saying "your token is no longer valid" when I know that I can refresh it. That it's also a valid credentials scenario IMO.